test/python/algorithms/optimizers/test_spsa.py

```diff
@@ -222,3 +222,30 @@ def test_qnspsa_fidelity_primitives(self):
             result = fidelity(initial_point, initial_point)
 
             self.assertAlmostEqual(result[0], 1)
+
+    def test_qnspsa_max_evals_grouped(self):
+        """Test using max_evals_grouped with QNSPSA."""
+        circuit = PauliTwoDesign(3, reps=1, seed=1)
+        num_parameters = circuit.num_parameters
+        obs = Z ^ Z ^ I
+
+        estimator = Estimator()
+
+        initial_point = np.array(
+            [0.82311034, 0.02611798, 0.21077064, 0.61842177, 0.09828447, 0.62013131]
+        )
+
+        def objective(x):
+            x = np.reshape(x, (-1, num_parameters)).tolist()
+            n = len(x)
+            return estimator.run(n * [circuit], n * [obs.primitive], x).result().values.real
+
+        fidelity = QNSPSA.get_fidelity(circuit)
+        optimizer = QNSPSA(fidelity)
+        optimizer.maxiter = 1
+        optimizer.learning_rate = 0.05
+        optimizer.perturbation = 0.05
+        optimizer.set_max_evals_grouped(50)  # greater than 1
+
+        result = optimizer.minimize(objective, initial_point)
+        self.assertIsNotNone(result)
```

Comment:
Could we seed the estimator (just avoiding any possible stochastic convergence failures), and with that, is there anything sensible we can test about the qualities of the output results?  It might be good to test that there are the correct number of them, and that they're within the right ballpark - we can do the quantitative check only if there's a seed, though.